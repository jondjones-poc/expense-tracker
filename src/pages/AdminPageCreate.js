import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm';
import { startAddExpense } from '../actions/expenseActions';

export class AdminPageCreate extends React.Component {
  constructor(props) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
  };
  onSubmit(expense) {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container" >
            <h1 className="page-header__title">
              Add Expense
            </h1>
          </div>
        </div>
        <div className="content-container" >
            <ExpenseForm
              history={this.props.history} 
              onSubmit={this.onSubmit}
            />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AdminPageCreate);
