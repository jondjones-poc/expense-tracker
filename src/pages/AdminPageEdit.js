import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../components/ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenseActions';

export class AdminPageEdit extends React.Component {
    constructor(props) {
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
        this.onRemove = this.onRemove.bind(this);
    };
    onSubmit(expense) {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove() {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container" >
                        <h1 className="page-header__title">
                            Edit Expense
                        </h1>
                    </div>
                </div>
                <div className="content-container" >
                        <ExpenseForm
                                expense={this.props.expense}
                                onSubmit={this.onSubmit}
                            />
                            <button onClick={this.onRemove} className="button button-remove">
                                Remove Expense
                            </button>
                    </div>
            </div>
        );
    }
}
 
const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  });
  
  const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
  });

export default connect(mapStateToProps, mapDispatchToProps)(AdminPageEdit);