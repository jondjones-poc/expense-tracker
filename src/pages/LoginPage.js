import React from 'react'
import { connect } from 'react-redux';
import { startLogin } from '../actions/authActions';

const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 class="box-layout__title">
                Expense Tracker
            </h1>
            <p>Track your expenses... <br />like a BOSS</p>
            <button onClick={startLogin} className="button">Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);