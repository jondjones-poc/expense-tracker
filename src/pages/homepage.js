import React from 'react'
import ReactDOM from 'react-dom'
import ExpenseList from '../components/expenseList';
import ExpenseListFilters from '../components/expenseListFilters';
import ExpenseSummary from '../components/ExpenseSummary';

const Homepage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default Homepage;