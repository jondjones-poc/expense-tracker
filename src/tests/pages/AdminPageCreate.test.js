import React from 'react';
import { shallow } from 'enzyme';
import { AdminPageCreate }  from '../../pages/AdminPageCreate';
import expensesTestData from '../Fixtures/expensesTestdata';

let startAddExpense, history, wrapper;

beforeEach(() => {
    startAddExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <AdminPageCreate 
            startAddExpense={startAddExpense} 
            history={history} 
        />);
});

test('should render admin page create', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle submit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expensesTestData[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startAddExpense).toHaveBeenLastCalledWith(expensesTestData[1]);
});