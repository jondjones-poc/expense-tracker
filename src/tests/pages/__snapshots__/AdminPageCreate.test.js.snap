// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render admin page create 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      history={
        Object {
          "push": [MockFunction],
        }
      }
      onSubmit={[Function]}
    />
  </div>
</div>
`;
